source/ota.c

```diff
@@ -2584,7 +2584,7 @@ static IngestResult_t decodeAndStoreDataBlock( OtaFileContext_t * pFileContext,
     }
 
     /* Decode the file block if space is allocated. */
-    if( payloadSize > 0u )
+    if( ( payloadSize > 0u ) && ( payloadSize >= messageSize ) )
```

Comment:
1. Is it always an error if the payloadSize is not greater than or equal to messageSize? What is the expected behavior here? I'm wondering where messageSize comes from and if it could be smaller in some cases, for example the last block in a download.
2. This check is redundant, if we are requiring payloadSize to be >= message size, then we don't need to also check that it's greater than 0. We can just check that it's greater than or equal to the messageSize (assuming this is what we want to be checking).

Reply:
1. Yes, It's always an error if the messageSize is greater than the payloadSize. 
2. I will change it in the next commit.